xonsh/built_ins.py

```diff
@@ -720,6 +724,7 @@ def load_builtins(execer=None, config=None, login=False, ctx=None):
     builtins.__xonsh_execer__ = execer
     builtins.__xonsh_commands_cache__ = CommandsCache()
     builtins.__xonsh_all_jobs__ = {}
+    builtins.__xonsh_enabled_builtin_commands__ = set(all_builtin_commands.keys())
```

Comment:
While I am saying that it is somewhat confusing that enable wouldn't on aliases, I believe it to be much more confusing that disable/del wouldn't work on aliases and builtin commands exactly the same. Additionally, that it could be possible to have a builtin and an alias with the same name, is also harder to explain and understand the rules for. 

I guess I am in favor of keeping the core model as simple as possible: one environment, one place where commands go, one interface to color management, etc. I think the benefits to teaching the language for keeping things simple are hard for me to understate.  It is late so the only counter example I can think of is something like if Python had `sys.modules` for most module imports and then another `sys.builtin_modules` for thing which Python itself loaded.  As another, potentially bad example, C++ does something like this with namespaces, where anyone is free to modify any namespace - except `std` which is somehow sacred.

Again, I am happy to put in a PR that changes the name or aliases and related code to something more generic and semantically meaningful to the full suite of uses cases it has been asked to serve. 

I guess another option would be to make aliases, or whatever we call it, a [ChainMap](https://docs.python.org/3/library/collections.html#collections.ChainMap) or similar. This would allow a distinction between command types, and different references to them.  User commands could live in the first map of the chain, these builtins would be the second, etc. But fundamentally, the `run_subproc()` function would only look things up in one mapping, users would only need to know about one mapping, etc.

I hope that some this makes sense :). I'd like for all of us to be happy with the underlying model, too!